certbot/cli.py

```diff
@@ -1567,16 +1567,17 @@ def parse_preferred_challenges(pref_challs):
     # Ignore tls-sni-01 from the list, and generates a deprecation warning
     # TODO: remove this option completely in few releases
     if "tls-sni-01" in challs:
-        sys.stderr.write('Use of tls-sni-01 in preferred-challenges is '
-                         'deprecated, and will be removed soon.\n')
-        challs.remove("tls-sni-01")
+        logger.warning('TLS-SNI-01 support is deprecated. This value is being dropped from the '
+                       'setting of --preferred-challenges and future versions of Certbot will '
+                       'error if it is included.')
+        challs = [chall for chall in challs if chall != "tls-sni-01"]
 
     unrecognized = ", ".join(name for name in challs
                              if name not in challenges.Challenge.TYPES)
     if unrecognized:
         raise errors.Error(
             "Unrecognized challenges: {0}".format(unrecognized))
-    return challs
+    return list(challs)
```

Comment:
nit: Now that we've gone back to a `list` from a `set`, I don't think the change here is necessary.

Reply:
Done.